libimage/manifest_list_test.go

```diff
@@ -75,3 +75,52 @@ func TestCreateAndTagManifestList(t *testing.T) {
 	// Both origin list and newly tagged list should point to same image id
 	require.Equal(t, image.ID(), taggedImage.ID())
 }
+
+// Following test ensure that we test  Removing a manifestList
+// Test tags two manifestlist and deletes one of them and
+// confirms if other one is not deleted.
+func TestCreateAndRemoveManifestList(t *testing.T) {
+
+	tagName := "manifestlisttagged"
+	listName := "manifestlist"
+	runtime, cleanup := testNewRuntime(t)
+	defer cleanup()
+	ctx := context.Background()
+
+	list, err := runtime.CreateManifestList(listName)
+	require.NoError(t, err)
+	require.NotNil(t, list)
+
+	manifestListOpts := &ManifestListAddOptions{All: true}
+	_, err = list.Add(ctx, "docker://busybox", manifestListOpts)
+	require.NoError(t, err)
+
+	list, err = runtime.LookupManifestList(listName)
+	require.NoError(t, err)
+	require.NotNil(t, list)
+
+	lookupOptions := &LookupImageOptions{ManifestList: true}
+	image, _, err := runtime.LookupImage(listName, lookupOptions)
+	require.NoError(t, err)
+	require.NotNil(t, image)
+	err = image.Tag(tagName)
+	require.NoError(t, err, "tag should have succeeded: %s", tagName)
+
+	taggedImage, _, err := runtime.LookupImage(tagName, lookupOptions)
+	require.NoError(t, err)
+	require.NotNil(t, taggedImage)
+	// Both origin list and newly tagged list should point to same image id
+	require.Equal(t, image.ID(), taggedImage.ID())
+
+	// Try deleting the manifestList with tag
+	rmReports, rmErrors := runtime.RemoveImages(ctx, []string{"manifestlisttagged"}, &RemoveImagesOptions{Force: true, LookupManifest: true})
+	require.Nil(t, rmErrors)
+	require.Equal(t, []string{"localhost/manifestlisttagged:latest"}, rmReports[0].Untagged)
+
+	// Original tag should still exists
+	imageRe, _, err := runtime.LookupImage(listName, lookupOptions)
+	require.NoError(t, err)
+	require.NotNil(t, imageRe)
+	require.Equal(t, image.ID(), imageRe.ID())
```

Comment:
For the sake of completeness we should now remove `listName` and make sure it's marked as deleted.